src/com/twitter/intellij/pants/components/impl/FastpassUpdater.java

```diff
@@ -58,18 +61,59 @@
   private static class FastpassData {
     public String version;
     public String projectName;
+
     public FastpassData(String version, String projectName) {
       this.version = version;
       this.projectName = projectName;
     }
   }
 
+  public static final class Action extends AnAction {
+
+    public static final String DIALOG_TITLE = "Fastpass Update";
+
+    @Override
+    public void update(@NotNull AnActionEvent e) {
+      boolean show = fastpassBinaryExists();
+      e.getPresentation().setEnabledAndVisible(show);
+    }
+
+    @Override
+    public void actionPerformed(@NotNull AnActionEvent e) {
+      Project project = e.getProject();
+      if (fastpassBinaryExists()) {
+        systemVersion(FASTPASS_PATH)
+          .ifPresent(systemVersion -> extractFastpassData(project)
+            .ifPresent(data -> {
+              if (!data.version.equals(systemVersion)) {
+                String message = "Do you want to update fastpass to version: " + systemVersion + "?";
+                int answer = Messages.showYesNoDialog(project, message, DIALOG_TITLE, null);
+                if (answer == Messages.YES) {
+                  updateFastpassVersion(project, data);
+                }
+              } else {
+                Messages.showInfoMessage(project, "Fastpass is already up to date", DIALOG_TITLE);
+              }
+            }));
+      }
+    }
+
+    private boolean fastpassBinaryExists() {
+      VirtualFile fastpassBinary = LocalFileSystem.getInstance().findFileByPath(FASTPASS_PATH);
+      return fastpassBinary != null && fastpassBinary.exists();
+    }
+  }
+
   public static void initialize() {
     if (!initialized) {
       synchronized (FastpassUpdater.class) {
         if (!initialized) {
           ScheduledExecutorService timer = Executors.newScheduledThreadPool(1);
```

Comment:
Just to confirm, does this check happen for users who are not using a BSP project?

Reply:
Good catch, I lost this logic somewhere on the way. So it would always start the thread, but not do anything than check for open bsp projects once a day. I updated it, so that the thread is started after opening first bsp project (rather than any project).